database/migrations/0061-event_refactoring.ts

```diff
@@ -0,0 +1,56 @@
+/* MIGRATION TO REFACTOR THE EVENT_PROTOCOL TABLE
+ *
+ * This migration refactors the `event_protocol` table.
+ * It renames the table to `event`, introduces new fields and renames others.
+ */
+
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+export async function upgrade(queryFn: (query: string, values?: any[]) => Promise<Array<any>>) {
+  await queryFn('RENAME TABLE `event_protocol` TO `events`;')
+  await queryFn('ALTER TABLE `events` RENAME COLUMN `user_id` TO `affected_user_id`;')
+
+  await queryFn(
+    'ALTER TABLE `events` ADD COLUMN `acting_user_id` int(10) unsigned DEFAULT NULL AFTER `affected_user_id`;',
+  )
+  await queryFn('UPDATE `events` SET `acting_user_id` = `affected_user_id`;')
+  await queryFn(
+    'ALTER TABLE `events` MODIFY COLUMN `acting_user_id` int(10) unsigned NOT NULL AFTER `affected_user_id`;',
+  )
+
+  await queryFn('ALTER TABLE `events` RENAME COLUMN `x_user_id` TO `involved_user_id`;')
+  await queryFn('ALTER TABLE `events` DROP COLUMN `x_community_id`;')
+  await queryFn('ALTER TABLE `events` RENAME COLUMN `transaction_id` TO `involved_transaction_id`;')
+  await queryFn(
+    'ALTER TABLE `events` RENAME COLUMN `contribution_id` TO `involved_contribution_id`;',
+  )
+  await queryFn(
+    'ALTER TABLE `events` MODIFY COLUMN `message_id` int(10) unsigned DEFAULT NULL AFTER `involved_contribution_id`;',
+  )
+  await queryFn(
+    'ALTER TABLE `events` RENAME COLUMN `message_id` TO `involved_contribution_message_id`;',
+  )
+
```

Comment:
please refactor the created_at column by adding milliseconds
...\`created_at\` datetime(3) NOT NULL DEFAULT CURRENT_TIMESTAMP(3)